haskell/test/SMT/Model/BaseDAO/Types.hs

```diff
@@ -0,0 +1,269 @@
+-- SPDX-FileCopyrightText: 2021 TQ Tezos
+-- SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-missing-signatures #-}
+
+module SMT.Model.BaseDAO.Types
+  ( ModelT (..)
+  , ModelState (..)
+  , SimpleContractType (..)
+  , SimpleFA2Contract (..)
+  , OtherContract (..)
+  , SimpleOperation (..)
+  , runModelT
+  , getStore
+  , getConfig
+  , modifyStore
+  , execOperations
+  , isFA2Contract
+  , mkSimpleOperation
+  , (-$$?)
+
+  , ContractType (..)
+  , ModelCall (..)
+  , ModelError (..)
+  , contractErrorToModelError
+  ) where
+
+import Universum
+
+import Control.Monad.Except (MonadError, throwError)
+import Data.Typeable (cast)
+import qualified Data.Map as M
+import Fmt
+
+import Michelson.Interpret
+import Michelson.Text
+import qualified Michelson.Typed as T
+import Michelson.Typed.Value
+import Lorentz hiding (not, cast, get)
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import Tezos.Address (ContractHash(..))
+import Tezos.Core
+
+import Ligo.BaseDAO.Types
+
+-- | Transformer used in haskell implementation of BaseDAO
+newtype ModelT a = ModelT
+  { unModelT :: (ExceptT ModelError $ State ModelState) a
+  } deriving newtype (Functor, Applicative, Monad, MonadState ModelState, MonadError ModelError)
+
+-- | A type that keep track of blockchain state
+data ModelState = ModelState
+  { msFullStorage :: FullStorage
+
```

Comment:
I see you have added a `ContractType` and used it in `applyCallCustom`, I wonder why you didn't instead go with my suggestion in the [previous comment](https://github.com/tqtezos/baseDAO/pull/282#discussion_r662543981) to add a `Map` with functions here to use instead of the lambdas:
```suggestion
  , msCustomEps :: Map MText (Bytestring -> ModelT ())
```

Reply:
Make sense, I've updated this now, also added all the other lambda as part of `ModelState`.